tf2onnx/rewriter/cond_rewriter.py

```diff
@@ -139,150 +173,112 @@ def _cut_off_connection(self, cond_context):
-                    # true branch
-                    else:
-                        true_graph_context.union(downstream_graph[node])
-                    switchs.add(input_node)
-                    # self._workaround_for_placeholder(input_node.input[0])
-                else:
-                    downstream_graph[input_node].nodes.add(input_node)
-                    downstream_graph[input_node].union(downstream_graph[node])
-                    stack.append(input_node)
-        if true_graph_context.empty() and false_graph_context.empty():
+        def stop_at_switch(node):
+            if self._is_switch(node):
+                switchs.add(node)
+                return False
+            return True
+        branch_nodes_1 = self.g.extract_sub_graph_nodes(
+            [merge_input_1],
+            stop_at_switch
+        )
+        branch_nodes_2 = self.g.extract_sub_graph_nodes(
+            [merge_input_2],
+            stop_at_switch
+        )
+        branch_type_1 = self._branch_type(merge_input_1, branch_nodes_1)
+        branch_type_2 = self._branch_type(merge_input_2, branch_nodes_2)
+        # handle the case one branch only contains a const node
+        if branch_type_1 == BranchType.UNKNOWN and branch_type_2 == BranchType.UNKNOWN:
             raise ValueError("Cannot handle the case both true and false branchs only \
                              contain const nodes for now.")
-        for graph_context in non_input_node_downstream_graph:
-            if graph_context.same_branch(true_graph_context) or \
-                    true_graph_context.empty():
-                true_graph_context.union(graph_context)
-            else:
-                false_graph_context.union(graph_context)
-        # one node cannot belong to both true and false graph
-        intersection = true_graph_context.intersection(false_graph_context)
-        if intersection:
-            raise ValueError("true graph and false graph intersect at [{}]".format(
-                ",".join(intersection)
-            ))
-        log.debug("=================false body graph===============")
-        log.debug(false_graph_context.nodes)
-        log.debug(false_graph_context.output)
-        log.debug("=================true body graph===============")
-        log.debug(true_graph_context.nodes)
-        log.debug(true_graph_context.output)
-        return true_graph_context, false_graph_context, switchs
+        if branch_type_1 == branch_type_2:
+            raise ValueError("true graph and false graph are intersected")
+        if branch_type_1 == BranchType.TRUE:
+            true_branch_nodes = branch_nodes_1
+            true_output = merge_input_1
+            false_branch_nodes = branch_nodes_2
+            false_output = merge_input_2
+        else:
+            true_branch_nodes = branch_nodes_2
+            true_output = merge_input_2
+            false_branch_nodes = branch_nodes_1
+            false_output = merge_input_1
```

Comment:
how would you handle cases where one branch type is unknown, another one is true or false. Is this is supported? If not , throw exception please